docs/M365/MDO/ResendFailedMail.md

```diff
@@ -0,0 +1,130 @@
+
+- Use the `client_id`, `tenant_id`, and `client_secret` obtained during app registration to authenticate with Microsoft Graph in the script (connection instructions below).
+
+<br><br>
+4. After completion of the above steps, and before running the script, connect to Exchange Online and Graph API with Powershell, as follows - make sure to insert your values into the connection string for Graph:
+
+```powershell
+Connect-ExchangeOnline -ShowBanner:$false
+
+Connect-MgGraph -TenantId "[YOUR TENANT ID HERE]" -ClientSecretCredential (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "[YOUR APP ID HERE]", (ConvertTo-SecureString -String "[VALUE FIELD OF YOUR SECRET HERE]" -AsPlainText -Force)) -NoWelcome
+```
+
+You can find the Microsoft Graph modules in the following link:<br>
+&nbsp;&nbsp;&nbsp;&nbsp;https://www.powershellgallery.com/packages/Microsoft.Graph/<br>
+
+&nbsp;&nbsp;&nbsp;&nbsp;https://learn.microsoft.com/en-us/powershell/microsoftgraph/installation?view=graph-powershell-1.0#installation
+
+
+You can find the Exchange module and information in the following links:<br>
+&nbsp;&nbsp;&nbsp;&nbsp;https://learn.microsoft.com/en-us/powershell/exchange/exchange-online-powershell-v2?view=exchange-ps<br>
+&nbsp;&nbsp;&nbsp;&nbsp;https://www.powershellgallery.com/packages/ExchangeOnlineManagement
+<br><br>
+
+## Parameters and Use Cases:
+Run the script with the Days parameter to specify the number of days in the past to retrieve email with a **Failed** status as well as with the Sender parameter. You will be prompted before executing this command.
+
+> [!IMPORTANT]
+> Make sure the original cause of the failed sending is fixed, or the script will also fail to send it.
+
+!['No Logical inconsistencies found'](img/ResendFailedMail-4Days+Sender.png)
+
+**Script Output 1: Resending Last 4 Days of Failed Email from Specific Sender**
+
+<br><br>
+Run the script with no parameters to resend all Failed email from the past day.
+
+!['No Logical inconsistencies found'](img/ResendFailedMail-No_Params.png)
+
+**Script Output 2: Default Execution of Script with No Parameters**
+
+
+## Additional examples
+
+To resend email from specific sender, recipient, and number of days, run the following:<br>
+```powershell
+.\ResendFailedMail.ps1 -Sender gary@contoso.com -Recipient ahmad@fabrikam.com -Days 7
+```
+
+To resend email from a specific sender for the past 5 days without a confirmation prompt, run the following:<br>
+```powershell
+.\ResendFailedMail.ps1 -Force -Sender gary@contsoso.com -Days 5
+```
+
+## Parameters - all parameters are optional
+
+Parameter | Description |
+----------|-------------|
+Sender | Filter emails based on the sender's address.
+Recipient | Filter emails based on the recipient's address.
+Subject | Filter emails based on the email Subject.
+MessageId | Filter emails based on the MessageId address. **You must put the MessageId in double quotes**
```

Comment:
Why does the `MessageId` need to be in double quotes?

Reply:
This is documented and I have confirmed it tests: 
https://learn.microsoft.com/en-us/powershell/module/exchange/get-messagetrace?view=exchange-ps#-messageid 

![image](https://github.com/user-attachments/assets/445b0d8d-6b09-4905-b162-92e4d10dbd85)

Reply:
That is for the cmdlet itself, which you don't provide the `MessageID` parameter when running the command `Get-MessageTrace`. You are just filtering it afterwards and it isn't required, as you defined it as a string array already, so it converts it to a string, which is all the quotation marks are doing. 

![image](https://github.com/user-attachments/assets/35f1a793-aa12-4f19-ba25-9511a98be1ae)

Reply:
this is about double quote for message ids:
It fails without quotes and works with double quotes:
![image](https://github.com/user-attachments/assets/61d874b8-ba3d-466d-8053-aa817539a913)

Not only in the script in the original cmdlet as well:
![image](https://github.com/user-attachments/assets/6ac1ea2e-c858-4c90-ae17-0068ac333db9)

Reply:
Didn't know that PowerShell treats that differently. Good to know.